saleor/core/utils/random_data.py

```diff
@@ -82,11 +85,15 @@ def create_attributes_and_values(schema, attribute_key):
 
 
 def create_product_class_with_attributes(name, schema):
-    product_class = get_or_create_product_class(name=name)
+    product_attributes_schema = schema.pop('product_attributes', {})
+    variant_attributes_schema = schema.pop('variant_attributes', {})
```

Comment:
You're modifying the object that is passed in which means this function has potentially unexpected side effects.